DTXManiaプロジェクト/コード/スコア、曲/CDTX.cs

```diff
@@ -1531,135 +1507,66 @@ public void tWavの再生停止( int nWaveの内部番号, bool bミキサーか
-					//    }
-					//}
-
-					// まず1つめを登録する
-					try
-					{
-						cwav.rSound[ 0 ] = CDTXMania.Sound管理.tサウンドを生成する( str );
-						cwav.rSound[ 0 ].n音量 = 100;
-						if ( !CDTXMania.ConfigIni.bDynamicBassMixerManagement )
-						{
-							cwav.rSound[ 0 ].tBASSサウンドをミキサーに追加する();
-						}
-						if ( CDTXMania.ConfigIni.bLog作成解放ログ出力 )
-						{
-							Trace.TraceInformation( "サウンドを作成しました。({3})({0})({1})({2}bytes)", cwav.strコメント文, str, cwav.rSound[ 0 ].nサウンドバッファサイズ, cwav.rSound[ 0 ].bストリーム再生する ? "Stream" : "OnMemory" );
-						}
-					}
-					catch ( Exception e )
-					{
-						cwav.rSound[ 0 ] = null;
-						Trace.TraceError( "サウンドの作成に失敗しました。({0})({1})", cwav.strコメント文, str );
-						Trace.TraceError( "例外: " + e.Message );
-					}
-
-					#region [ 同時発音数を、チャンネルによって変える ]
-					int nPoly = nPolyphonicSounds;
-					if ( CDTXMania.Sound管理.GetCurrentSoundDeviceType() != "DirectSound" )	// DShowでの再生の場合はミキシング負荷が高くないため、
-					{																		// チップのライフタイム管理を行わない
-						if      ( cwav.bIsBassSound )   nPoly = (nPolyphonicSounds >= 2)? 2 : 1;
-						else if ( cwav.bIsGuitarSound ) nPoly = (nPolyphonicSounds >= 2)? 2 : 1;
-						else if ( cwav.bIsSESound )     nPoly = 1;
-						else if ( cwav.bIsBGMSound)     nPoly = 1;
-					}
-					if ( cwav.bIsBGMSound ) nPoly = 1;
-					#endregion
-
-					// 残りはClone等で登録する
-					//if ( bIsDirectSound )	// DShowでの再生の場合はCloneする
-					//{
-					//    for ( int i = 1; i < nPoly; i++ )
-					//    {
-					//        cwav.rSound[ i ] = (CSound) cwav.rSound[ 0 ].Clone();	// #24007 2011.9.5 yyagi add: to accelerate loading chip sounds
-					//        // CDTXMania.Sound管理.tサウンドを登録する( cwav.rSound[ j ] );
-					//    }
-					//    for ( int i = nPoly; i < nPolyphonicSounds; i++ )
-					//    {
-					//        cwav.rSound[ i ] = null;
-					//    }
-					//}
-					//else															// WASAPI/ASIO時は通常通り登録
-					{
-						for ( int i = 1; i < nPoly; i++ )
-						{
-							try
-							{
-								cwav.rSound[ i ] = CDTXMania.Sound管理.tサウンドを生成する( str );
-								cwav.rSound[ i ].n音量 = 100;
-								if ( !CDTXMania.ConfigIni.bDynamicBassMixerManagement )
-								{
-									cwav.rSound[ i ].tBASSサウンドをミキサーに追加する();
-								}
-								if ( CDTXMania.ConfigIni.bLog作成解放ログ出力 )
-								{
-									Trace.TraceInformation( "サウンドを作成しました。({3})({0})({1})({2}bytes)", cwav.strコメント文, str, cwav.rSound[ 0 ].nサウンドバッファサイズ, cwav.rSound[ 0 ].bストリーム再生する ? "Stream" : "OnMemory" );
-								}
-							}
-							catch ( Exception e )
-							{
-								cwav.rSound[ i ] = null;
-								Trace.TraceError( "サウンドの作成に失敗しました。({0})({1})", cwav.strコメント文, str );
-								Trace.TraceError( "例外: " + e.Message );
-							}
-						}
-					}
-				}
-				catch( Exception exception )
-				{
-					Trace.TraceError( "サウンドの生成に失敗しました。({0})({1})({2})", exception.Message, cwav.strコメント文, str );
-					for ( int j = 0; j < nPolyphonicSounds; j++ )
-					{
-						cwav.rSound[ j ] = null;
-					}
-					//continue;
-				}
-			}
-		}
-		public static string tZZ( int n )
+	    {
```

Comment:
This method had duplicate logic for processing cwav.rSound members, depending on the index.